packages/devtools_app/lib/src/screens/vm_developer/vm_developer_common_widgets.dart

```diff
@@ -424,13 +424,25 @@ class RetainingPathWidget extends StatelessWidget {
   /// parentMapKey, and parentField where applicable.
   String _retainingObjectDescription(RetainingObject object) {
     if (object.parentListIndex != null) {
-      final ref = object.value as InstanceRef;
-      return 'Retained by element [${object.parentListIndex}] of ${ref.classRef?.name ?? '<parentListName>'}';
+      final ref = object.value;
+      if (ref is InstanceRef) {
+        return 'Retained by element [${object.parentListIndex}] of'
+            ' ${ref.classRef?.name ?? '<parentListName>'}';
+      } else {
+        return 'Retained by element [${object.parentListIndex}] of '
+            '${_objectName(ref) ?? '<unknown>'}';
+      }
     }
 
     if (object.parentMapKey != null) {
-      final ref = object.value as InstanceRef;
-      return 'Retained by element at [${_objectName(object.parentMapKey)}] of ${ref.classRef?.name ?? '<parentMapName>'}';
+      final ref = object.value;
+      if (ref is InstanceRef) {
+        return 'Retained by element at [${_objectName(object.parentMapKey)}] of'
```

Comment:
Pull `_objectName(object.parentMapKey)` into a local and use it in both the conditions below. We'll want to do the same with the `parentListIndex` above and `inboundRef.parentListIndex` below.

Reply:
Done